File: backend/src/main/java/com/stockanalysis/service/FinancialDataService.java
package com.stockanalysis.service;

import com.stockanalysis.dto.FinancialDataDTO;
import com.stockanalysis.entity.enums.PeriodType;
import java.util.List;

public interface FinancialDataService {
    FinancialDataDTO createOrUpdate(FinancialDataDTO financialDataDTO);
    FinancialDataDTO getById(Long id);
    void delete(Long id);
    List<FinancialDataDTO> getByStockId(Long stockId);
    List<FinancialDataDTO> getByStockAndPeriodType(Long stockId, PeriodType periodType);
    FinancialDataDTO getByStockAndPeriod(Long stockId, PeriodType periodType, Integer year, Integer quarter);
}

File: frontend/src/services/financialDataService.js
import api from './api';

const financialDataService = {
  getById: (id) => {
    return api.get(`/financial-data/${id}`);
  },

  getByStockId: (stockId) => {
    return api.get(`/financial-data/stock/${stockId}`);
  },

  getByStockAndPeriodType: (stockId, periodType) => {
    return api.get(`/financial-data/stock/${stockId}/period/${periodType}`);
  },

  createOrUpdate: (financialData) => {
    return api.post('/financial-data', financialData);
  },

  delete: (id) => {
    return api.delete(`/financial-data/${id}`);
  }
};

export default financialDataService;
